### PR TITLE
config: fix TestGetConfigFromContext on Windows

### LINKED ISSUE #3305: Failing `TestGetConfigFromContext` on Windows
Path-related issue introduced in #3206.
```
2024-02-06T09:12:48.5112738Z === RUN   TestGetConfigFromContext/relative-path
2024-02-06T09:12:48.5113506Z     server_test.go:60: 
2024-02-06T09:12:48.5114548Z         	Error Trace:	D:/a/neo-go/neo-go/cli/server/server_test.go:60
2024-02-06T09:12:48.5115419Z         	Error:      	Not equal: 
2024-02-06T09:12:48.5116507Z         	            	expected: "../../config/chains/testnet"
2024-02-06T09:12:48.5134020Z         	            	actual  : "..\\..\\config\\chains\\testnet"
2024-02-06T09:12:48.5134736Z         	            	
2024-02-06T09:12:48.5135258Z         	            	Diff:
2024-02-06T09:12:48.5135970Z         	            	--- Expected
2024-02-06T09:12:48.5136619Z         	            	+++ Actual
2024-02-06T09:12:48.5137312Z         	            	@@ -1 +1 @@
2024-02-06T09:12:48.5138197Z         	            	-../../config/chains/testnet
2024-02-06T09:12:48.5139081Z         	            	+..\..\config\chains\testnet
2024-02-06T09:12:48.5140004Z         	Test:       	TestGetConfigFromContext/relative-path
2024-02-06T09:12:48.5140787Z --- FAIL: TestGetConfigFromContext (0.01s)
2024-02-06T09:12:48.5141644Z     --- PASS: TestGetConfigFromContext/config-path (0.01s)
2024-02-06T09:12:48.5142686Z     --- PASS: TestGetConfigFromContext/config-file (0.00s)
2024-02-06T09:12:48.5143639Z     --- FAIL: TestGetConfigFromContext/relative-path (0.00s)
```

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Failing test caused by hardcoded relative path check which is different for Windows.

Close #3305

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -114,7 +114,8 @@ func LoadFile(configPath string, relativePath ...string) (Config, error) {
 // updateRelativePaths updates relative paths in the config structure based on the provided relative path.
 func updateRelativePaths(relativePath string, config *Config) {
 	updatePath := func(path *string) {
-		if *path != "" && !filepath.IsAbs(*path) {
+		unifiedPath := *path
+		if *path != "" && !filepath.IsAbs(*path) && unifiedPath[0] != '/' {
```
Comment (AnnaShaleva) [inline pkg/config/config.go:118]: Why do we need the latter condition?

Comment (AliceInHunterland) [reply pkg/config/config.go:118]: because on Windows filepath.IsAbs("/cn_wallet.json") is false

Comment (AnnaShaleva) [reply pkg/config/config.go:118]: Create a separate test for Windows.

Comment (AnnaShaleva) [reply pkg/config/config.go:118]: You may add Windows-specific test as a separate file and use build constraint for it (https://pkg.go.dev/go/build#hdr-Build_Constraints) or check the platform in runtime: `if runtime.GOOS == "windows"`.

Comment (AnnaShaleva) [reply pkg/config/config.go:118]: Not resolved.

**Review by AnnaShaleva — COMMENTED** [review]
Do not use `test` prefix to commit/PRs, this prefix is too generic. Use module-specific prefix, because we have tests with the same name in different modules.

File: cli/server/server_test.go [hunk 2]
```diff
@@ -49,15 +51,38 @@ func TestGetConfigFromContext(t *testing.T) {
...
+		if runtime.GOOS != "windows" {
+			t.Skip("skipping Windows specific test")
+		}
+
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.String("relative-path", "..\\..\\config", "")
+		set.Bool("testnet", true, "")
+		dir, _ := os.Getwd()
+		fmt.Println(dir)
+		set.String("config-file", ".\\testdata\\protocol.testnet.windows.yml", "")
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:64]: This file is not attached.

File: cli/server/server_test.go [hunk 3]
```diff
@@ -49,15 +51,38 @@ func TestGetConfigFromContext(t *testing.T) {
...
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		cfg, err := options.GetConfigFromContext(ctx)
+		require.NoError(t, err)
+		require.Equal(t, filepath.Join("..", "..", "config", "chains", "testnet"), cfg.ApplicationConfiguration.DBConfiguration.LevelDBOptions.DataDirectoryPath)
+		require.Equal(t, "\\cn_wallet.json", cfg.ApplicationConfiguration.Consensus.UnlockWallet.Path)
+		require.Equal(t, "\\notary_wallet.json", cfg.ApplicationConfiguration.P2PNotary.UnlockWallet.Path)
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:70]: We need two kinds of paths to be present in this file: relative and absolute. Currently only relative paths are included.

File: cli/server/server_test.go [hunk 4]
```diff
@@ -49,15 +51,38 @@ func TestGetConfigFromContext(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, netmode.TestNet, cfg.ProtocolConfiguration.Magic)
 	})
-	t.Run("relative-path", func(t *testing.T) {
+	t.Run("relative-path-windows", func(t *testing.T) {
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:54]: This test is failing.

Comment (AnnaShaleva) [inline cli/server/server_test.go:54]: "relative-path windows"? `relative-path` is literally a CLI option.

File: cli/server/server_test.go [hunk 5]
```diff
@@ -49,15 +51,38 @@ func TestGetConfigFromContext(t *testing.T) {
...
+		if runtime.GOOS != "windows" {
+			t.Skip("skipping Windows specific test")
+		}
+
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.String("relative-path", "..\\..\\config", "")
+		set.Bool("testnet", true, "")
+		dir, _ := os.Getwd()
+		fmt.Println(dir)
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:63]: Don't need this `Println` and `os.Getwd`.

File: cli/server/server_test.go [hunk 6]
```diff
@@ -49,15 +51,38 @@ func TestGetConfigFromContext(t *testing.T) {
...
+		set.String("config-file", ".\\testdata\\protocol.testnet.windows.yml", "")
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		cfg, err := options.GetConfigFromContext(ctx)
+		require.NoError(t, err)
+		require.Equal(t, filepath.Join("..", "..", "config", "chains", "testnet"), cfg.ApplicationConfiguration.DBConfiguration.LevelDBOptions.DataDirectoryPath)
+		require.Equal(t, "C:\\someFolder\\cn_wallet.json", cfg.ApplicationConfiguration.Consensus.UnlockWallet.Path)
+		require.Equal(t, "C:\\someFolder\\notary_wallet.json", cfg.ApplicationConfiguration.P2PNotary.UnlockWallet.Path)
+	})
+
+	t.Run("relative-path-non-windows", func(t *testing.T) {
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:73]: "relative-path non-windows"?

File: cli/server/testdata/protocol.testnet.windows.yml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  VerifyTransactions: false
+  P2PSigExtensions: false
+  Hardforks:
+    Aspidochelone: 210000
+    Basilisk: 2680000
+
+ApplicationConfiguration:
+  SkipBlockVerification: false
+  # LogPath could be set up in case you need stdout logs to some proper file.
+  # LogPath: ".\\log\\neogo.log"
```
Comment (AnnaShaleva) [inline cli/server/testdata/protocol.testnet.windows.yml:48]: Can be removed.

File: cli/server/testdata/protocol.testnet.windows.yml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  DBConfiguration:
+    Type: "leveldb" #other options: 'inmemory','boltdb'
+    # DB type options. Uncomment those you need in case you want to switch DB type.
+    LevelDBOptions:
+      DataDirectoryPath: ".\\chains\\testnet"
+  #    BoltDBOptions:
+  #      FilePath: ".\\chains\\testnet.bolt"
```
Comment (AnnaShaleva) [inline cli/server/testdata/protocol.testnet.windows.yml:55]: Can be removed, as far as comments above. It's a test-specific configuration file, we likely will never change this file.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

**Review by AnnaShaleva — APPROVED** [review]
Good, now I'm convinced that this functionality works properly on Windows.

**Comment by AnnaShaleva** [comment]
@roman-khimov, may be reviewed & merged.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**